src/bind-all.ts

```diff
@@ -35,6 +35,85 @@ function getBinding(original: Binding, sharedOptions?: boolean | AddEventListene
   return binding;
 }
 
+export function bindAll<Target extends EventTarget, Type extends string>(
+  target: Target,
+  bindings: [Binding<Target, Type>],
+  sharedOptions?: boolean | AddEventListenerOptions,
+): UnbindFn;
+export function bindAll<Target extends EventTarget, Type1 extends string, Type2 extends string>(
+  target: Target,
+  bindings: [Binding<Target, Type1>, Binding<Target, Type2>],
+  sharedOptions?: boolean | AddEventListenerOptions,
+): UnbindFn;
+export function bindAll<
+  Target extends EventTarget,
+  Type1 extends string,
+  Type2 extends string,
+  Type3 extends string,
+>(
+  target: Target,
+  bindings: [Binding<Target, Type1>, Binding<Target, Type2>, Binding<Target, Type3>],
+  sharedOptions?: boolean | AddEventListenerOptions,
+): UnbindFn;
+export function bindAll<
+  Target extends EventTarget,
+  Type1 extends string,
+  Type2 extends string,
+  Type3 extends string,
+  Type4 extends string,
+>(
+  target: Target,
+  bindings: [
+    Binding<Target, Type1>,
+    Binding<Target, Type2>,
+    Binding<Target, Type3>,
+    Binding<Target, Type4>,
+  ],
+  sharedOptions?: boolean | AddEventListenerOptions,
+): UnbindFn;
+export function bindAll<
+  Target extends EventTarget,
+  Type1 extends string,
+  Type2 extends string,
+  Type3 extends string,
+  Type4 extends string,
+  Type5 extends string,
+>(
+  target: Target,
+  bindings: [
+    Binding<Target, Type1>,
+    Binding<Target, Type2>,
+    Binding<Target, Type3>,
+    Binding<Target, Type4>,
+    Binding<Target, Type5>,
+  ],
+  sharedOptions?: boolean | AddEventListenerOptions,
+): UnbindFn;
+export function bindAll<
+  Target extends EventTarget,
+  Type1 extends string,
+  Type2 extends string,
+  Type3 extends string,
+  Type4 extends string,
+  Type5 extends string,
+  Type6 extends string,
+>(
+  target: Target,
+  bindings: [
+    Binding<Target, Type1>,
+    Binding<Target, Type2>,
+    Binding<Target, Type3>,
+    Binding<Target, Type4>,
+    Binding<Target, Type5>,
+    Binding<Target, Type6>,
```

Comment:
Will the typing stop working after 6 options? Is there a way to support any amount?

Reply:
It won't stop working, but the event type will fall back to `Event`. This happens due to the reason that we need to know the `type` to resolve the event correctly. Therefore, we need to have a generic type for each element of the binding.   

I tried another way of doing it: I built a union of bindings from all `on*` properties, something like this:

```ts
| {
  type: 'abort',
  listener: Listener<UIEvent, Window>,
  ...
}
| ... 
| {
  type: 'mousemove',
  listener: Listener<MouseEvent, Window>,
  ...
}
```

And this implementation didn't require a function overload. However, if it wasn't working correctly with "unknown" events(e.g. `DOMContenetLoaded`). 

Because once I added a fallback case to the union (`{type: string, listener: Listener<Event, EventTareget>}`), the listener type started to be `any` for "known" events (e.g. 'mousemove'), since `type` was also always matching the fallback implementation.

So, right now I don't have any good idea about solving this problem, but I think that 6 handlers are more than enough. 
We could increase it up 8-10, but I don't think that there is any practical use case where you need more than this.

Reply:
So for now I try to look at it as a necessary evil 😂